migrate.go

```diff
@@ -594,7 +594,7 @@ func (m *Migrate) readUp(from int, limit int, ret chan<- interface{}) {
 
 			// reached end, and didn't apply any migrations
 			if limit > 0 && count == 0 {
-				ret <- os.ErrNotExist
+				ret <- ErrNoChange
```

Comment:
This error is returned when a limit has been specified. e.g. you know how many migrations you expect to be run. So `os.ErrNotExist` is the correct error to return in this case.

Reply:
You're right. However, I gave to say I've made this mistake because the documentation and examples was not clear enough to me, can I help on this?

Reply:
Yep, feel free to update the docs! The [FAQ](https://github.com/golang-migrate/migrate/blob/master/FAQ.md) might be a good place to start.